modules/core/src/lib/attribute-manager.js

```diff
@@ -117,6 +117,7 @@ export default class AttributeManager {
     this.gl = gl;
 
     this.attributes = {};
+    this.shaderAttributes = {};
```

Comment:
Naming: compositeAttributes/primitiveAttributes/...

Reply:
I think `shaderAttibutes` is more descriptive for the "real" attributes in the shader. Renaming what are currently called "attributes" in Deck is appealing, since it seems we don't want the layers thinking in terms of GLSL attributes. But that would be a significant (API breaking?) change for solely aesthetic reasons.

Reply:
> I think `shaderAttibutes` is more descriptive

I see your point. It just struck me that these are essentially "accessor objects" per luma.gl's (glTF-inspired) terminology. Maybe we could define it so that one `Attribute` can generate one or more "accessors", and we can have a method `getAccessors()`?

Reply:
> Renaming what are currently called "attributes" in Deck is appealing. But that would be a significant (API breaking?) for solely aesthetic reasons.

That depends. If we can come up with an evolution of the concepts that really makes sense and aligns between luma.gl and deck.gl, then I would see that as a major improvement to the maintainability of our code base, that will make future feature development and discussions much easier.  So I am basically encouraging further thought/discussion in this area :)

Not that much of the AttributeManager API is used outside layers, we could probably support a few deprecated methods for backwards compatibility.